### ISSUE TITLE
Feature suggestions

### ISSUE THREAD

**Issue by benjisoft** [ks888/LambStatus#14]
I have a lot of suggestions, so I thought it would be a good idea to combine them all here. 

- Component Grouping #10 

- Two factor authentication #13 

- AWS SNS intergration  #11 

- Custom colours/ styling changeable via a settings page  #12 

	- This could be done by having the S3 page trigger a lambda even in order to modify the CSS in S3

- SSO

	- E.g. With google/ facebook

**Comment by benjisoft** [comment]
You could use amazon SNS for emailing/ texting subscribers with incident updates

**Comment by ks888** [comment]
I see. Yes, it sounds good idea to use amazon SNS for that purpose. Thank you for explaining.

**Comment by kbariotis** [comment]
SNS receives a fixed email address upon subscription and we won't be able to set subscriber's email addresses on run-time.

**Comment by kaustubhmenon** [comment]
A small suggestion/question: 
- Is their a way to change the HTML Layout or just CSS. We'd love to customize the UI to our brand, without having to use AWS CloudFormation Designer. Thank you!

**Comment by ks888** [comment]
Thank you for asking. Sorry, you can change only the logo and the background color so far.

But it's great to support the feature to make CSS customizable. Can you tell me what kinds of changes you want to make (like font size, font color or margin size)?

**Comment by kaustubhmenon** [comment]
Thanks for taking the time to respond. The ability to change `font-family`, `font-size`, `font-color`, `padding` and `margins` would be more than appreciated. 

  Below is a list of customizations that would be amazing to have.
- Integrate a new CSS/SCSS file for manipulating the UI, without touching any core SCSS files. Could also be similar to Edit CSS in Wordpress (https://en.support.wordpress.com/custom-design/editing-css/) . While inspecting the structure I noticed that all class names are appended with a random string in the end which could be a possible blocker for targeting CSS classes?
- Google font integration would be great
- Ability to add external links to the footer or header. Would love to add links for documentation, support, etc.
- Manipulate the structure of Header, Main Container and Footer would be a added bonus if possible. Something similar to https://www.statuspage.io/features/customization. 

I know I’m asking for a lot, but i feel that the above features would make LambStatus Awesome-er. I’d love to help in any capacity to help implement customizations/theme options or any other UI related needs. Thank you!

**Comment by ks888** [comment]
> Integrate a new CSS/SCSS file for manipulating the UI, without touching any core SCSS files. 

StatusPage.io also has a similar feature https://help.statuspage.io/knowledge_base/topics/using-custom-css . Though the css class has a random prefix since CSS modules are used, maybe we can let the principal UI components have the id attribute and customize the design like this:
```
#container {
    width: 90%;
    max-width: 850px;
}
```

I've created the issue for this: https://github.com/ks888/LambStatus/issues/92

**Comment by ks888** [comment]
> Google font integration would be great

Does it mean the option to choose the font family of the Google fonts? That sounds interesting.

**Comment by ks888** [comment]
> Ability to add external links to the footer or header. Would love to add links for documentation, support, etc.
> Manipulate the structure of Header, Main Container and Footer would be a added bonus if possible. Something similar to https://www.statuspage.io/features/customization. 

Maybe we can support these customizations by having the feature to customize the header html and footer html. I've created the issue for this: #93

**Comment by kaustubhmenon** [comment]
> Does it mean the option to choose the font family of the Google fonts? That sounds interesting.

Yes, a dropdown with a list of google fonts to choose from in the settings page. Where the user can choose fonts for Page Header and Paragraphs(General Text).

**Comment by ks888** [comment]
Thank you for your reply. I've created the issue for this: https://github.com/ks888/LambStatus/issues/94
